### ISSUE TITLE
Zabbix 3.0 compatability

### ISSUE THREAD

**Issue by colttt** [cavaliercoder/libzbxpgsql#34]
Hello,

short question, is this modul compatible with Zabbix 3.0?

best regards 
Colttt

**Comment by cavaliercoder** [comment]
I haven't tested yet. Will try to in the next couple of days. 

Thanks for the question!

**Comment by colttt** [comment]
Hello,

it looks like that it dont work.. here the errormessage from zabbix agent (DebugLevel=5)

```
 23047:20160218:111435.159 In process_active_checks() server:'172.16.101.61' port:10051
 23047:20160218:111435.159 In PG_CONNECT()
 23047:20160218:111435.159 In pg_connect()
 23047:20160218:111435.159 Connecting to PostgreSQL with: host=127.0.0.1 port=5432 user=top_secret_user password=top_secret dbname=zabbix
 23047:20160218:111435.167 End of pg_connect()
 23047:20160218:111435.168 End of PG_CONNECT()
 23047:20160218:111435.168 In need_meta_update() key:pg.connect[host=127.0.0.1 port=5432 user=top_secret_user password=top_secret,zabbix]
 23047:20160218:111435.169 End of need_meta_update():FAIL
 23047:20160218:111435.169 End of process_active_checks()
[...]

 23047:20160218:111535.182 In process_active_checks() server:'172.16.101.61' port:10051
 23047:20160218:111535.182 In PG_CONNECT()
 23047:20160218:111535.182 In pg_connect()
 23047:20160218:111535.182 Connecting to PostgreSQL with: host=127.0.0.1 port=5432 user=top_secret_user password=top_secret dbname=zabbix
 23047:20160218:111535.190 End of pg_connect()
 23047:20160218:111535.190 End of PG_CONNECT()
 23047:20160218:111535.190 In need_meta_update() key:pg.connect[host=127.0.0.1 port=5432 user=top_secret_user password=top_secret,zabbix]
 23047:20160218:111535.191 End of need_meta_update():FAIL
 23047:20160218:111535.191 End of process_active_checks()
[...]

 23047:20160218:111835.914 In add_check() key:'pg.setting[host=127.0.0.1 port=5432 user=top_secret_user password=top_secret,zabbix,password_encryption]' refresh:300 lastlogsize:0 mtime:0
 23047:20160218:111835.914 End of add_check()
 23047:20160218:111835.914 In add_check() key:'pg.setting[host=127.0.0.1 port=5432 user=top_secret_user password=top_secret,zabbix,port]' refresh:300 lastlogsize:0 mtime:0
 23047:20160218:111835.915 End of add_check()
 23047:20160218:111835.915 End of parse_list_of_checks():SUCCEED
 23047:20160218:111835.915 End of refresh_active_checks():SUCCEED
 23047:20160218:111835.915 __zbx_zbx_setproctitle() title:'active checks #1 [processing active checks]'
 23047:20160218:111835.915 In process_active_checks() server:'172.16.101.61' port:10051
 23047:20160218:111835.915 In PG_CONNECT()
 23047:20160218:111835.915 In pg_connect()
 23047:20160218:111835.915 Connecting to PostgreSQL with: host=127.0.0.1 port=5432 user=top_secret_user password=top_secret dbname=zabbix
 23047:20160218:111835.924 End of pg_connect()
 23047:20160218:111835.924 End of PG_CONNECT()
 23047:20160218:111835.924 In need_meta_update() key:pg.connect[host=127.0.0.1 port=5432 user=top_secret_user password=top_secret,zabbix]
 23047:20160218:111835.924 End of need_meta_update():FAIL
 23047:20160218:111835.924 End of process_active_checks()
 23047:20160218:111835.924 In get_min_nextcheck()
```

**Comment by cavaliercoder** [comment]
I haven't had a chance to look yet, but I understand `need_meta_update()` should only apply to log, logrt and eventlog item types. Which item type are you using for `pg.connect` in your template?

**Comment by colttt** [comment]
Hello,

thats you original Template, i dont change anything..

**Comment by cavaliercoder** [comment]
I've found the issue but haven't fixed it yet. The module needs to be compiled with the Zabbix v3 headers. The libzbxpgsql source package includes headers from v2.4 which break in v3.

I'll close this issue once resolved.

**Comment by cavaliercoder** [comment]
See: https://support.zabbix.com/browse/ZBX-10428

**Comment by cavaliercoder** [comment]
Zabbix have confirmed the module will need to be compiled for v2 and v3 independently. See the 'Changes in modules' section of:
https://www.zabbix.com/documentation/3.0/manual/installation/upgrade_notes_300#daemon_changes

Hopefully they will re-introduce v1 module compatibility since libzbxpgsql does not use `log` item types.

**Comment by colttt** [comment]
ok, i think it can be closed?! it would be fine if you update you files/debs/rpms here: https://sourceforge.net/projects/libzbxpgsl/files/

**Comment by cavaliercoder** [comment]
Hopefully Zabbix will reintroduce API v1 compatibility in Zabbix v3 in [ZBX-10428](https://support.zabbix.com/browse/ZBX-10428). If so, it will save module vendors the trouble of publishing double the number of packages.

Once I know which way they are going I will either update the current packages to work with Zabbix v2 and v3 or publish separate packages for each.

**Comment by cavaliercoder** [comment]
To compile for Zabbix v3, simply replace all headers in `include/` with the header files included in Zabbix v3 sources.

**Comment by michaelwemoto** [comment]
Thanks cavaliercoder, Does this module support streaming replication monitiroing? I had been trying to get it working in Zabbix 3.0 and seeing the same errors as above.

**Comment by cavaliercoder** [comment]
I've created #37 to implement monitoring of streaming replication.

**Comment by Johnybbc** [comment]
Hi cavaliercoder,
i compiled module and replaced all header files in include/ but i see only blank values. Debian 8.4 and Zabbix agent 3.0.1. Can you publish DEB package for Zabbix 3.0? Thanks.

**Comment by cavaliercoder** [comment]
Ubuntu and RHEL family packages now available for Zabbix v3 from: https://sourceforge.net/projects/libzbxpgsl/files/Zabbix3/

**Comment by cavaliercoder** [comment]
Fixed in 1c1bb46. Zabbix headers must now be manually specified for the desired version to build.

Packages for v0.2.1 with Zabbix v3 are available from https://sourceforge.net/projects/libzbxpgsl/files/Zabbix3/

When v1.0.0 is released, packages will be published to target both versions of Zabbix.

Unified packages will be available once Zabbix have resolved ABI compatibility issues in a later version of Zabbix v3 via https://support.zabbix.com/browse/ZBX-10428

**Comment by cavaliercoder** [comment]
Packages have been moved from Source Forge to AWS S3. Please see: http://cavaliercoder.com/libzbxpgsql/download
